forge/ee/db/controllers/DeviceGroup.js

```diff
@@ -120,52 +124,89 @@ module.exports = {
             // otherwise, throw a friendly error message along with the original error
             throw new Error(`Failed to update device group membership: ${err.message}`)
         }
+        if (changeCount > 0) {
+            // clean up where necessary
+            // check to see if the group is now empty
+            const remainingDevices = await deviceGroup.deviceCount()
+            if (remainingDevices === 0) {
+                deviceGroup.targetSnapshotId = null
```

Comment:
```suggestion
                deviceGroup.targetSnapshotId = null
```

Wondering if setting the snapshot to null when the group is emptied is the right behaviour. A user could be in the middle of changing the devices in the group (albeit doing it in multiple steps) - and clearing the snapshot mid-way through the task could be unexpected.

Reply:
I don't _think_ it is of consequence. 

it just means the pipeline needs to be executed to make it re-sync.

The issue this was added for is adding devices to a _spare_ empty group (months/years later) and suddently they fire up running old (dangerous?) snapshots.

I feel quite strongly that we DONT actually provide a checkbox `[x] remove current snapshot` / `[x] apply current snapshot` - I think it is dangerous and suspect most (other) users will not like the behaviour we are adding at all


From the ISSUE https://github.com/FlowFuse/flowfuse/issues/3260#issuecomment-1866211727:

> 
> @MarianRaphael I believe it would be prudent to consider this:
> 
> Personally, I believe this should be done via an option when adding a device to a group since you may NOT actually want this to occur immediately and without warning that "things" could start moving/happening.
> 
> Consider the scenario:
> 
> You have x devices that you need to add to a device group BUT FIRST you need to update the source snapshot
> The current snapshot is NOT what needs to be deployed (because changes have not yet been made to accommodate the 10 new devices)
> You CANNOT/DO NOT want to deploy the new changes to the existing devices without the other devices being online/ready to receive the new snapshot.
> By simply adding a checkbox "[x] Auto Deploy current device group snapshot 'version 2.1'?" at the time of adding the device to a group can make this problem go away!